include/IrcCore/ircdebug_p.h

```diff
@@ -143,7 +143,7 @@ static bool irc_debug_enabled(IrcConnection* c, uint l)
         dbg_init = true;
     }
 
-    return l <= dbg_level && (dbg_name.isEmpty() || QRegularExpression(QRegularExpression::wildcardToRegularExpression(dbg_name), QRegularExpression::CaseInsensitiveOption).match(c->displayName()).hasMatch());
+    return l <= dbg_level && (dbg_name.isEmpty() || QRegularExpression(dbg_name+".*", QRegularExpression::CaseInsensitiveOption).match(c->displayName()).hasMatch());
```

Comment:
I'm sorry but I don't see the point in 1) forcing users to use more complicated regexp instead of simple wildcards, and 2) forcing it to a "starts with" rule instead of letting users decide. What if they want an "ends with" rule, which is currently possible? Also, the commit message says "fix" but what does this fix? All I can see is a behavior change.

Reply:
The point was to use a feature that is also available in older Qt but I was wrong since  QRegularExpression::wildcardToRegularExpression does more than just a regex with a wildcard like foobar* which would correspond to a regex foobar.*
I'll look how to handle that.

Reply:
Qt docs prior QRegularExpression::wildcardToRegularExpression(): https://doc.qt.io/archives/qt-5.11/qregularexpression.html#wildcard-matching

> rewriting a regular expression in wildcard syntax to a Perl-compatible regular expression is a **very easy task**

Then: https://github.com/qt/qtbase/blob/04ee5795cc31ee81fb0c27bf55d9e8f662995753/src/corelib/text/qregularexpression.cpp#L1931-L2010

:rofl:

Reply:
I'd prefer to restore the QRegExp version guarded by a QT_VERSION check: https://github.com/communi/libcommuni/pull/100

Reply:
@jpnurmi  Like I done in the most recent commit?